### ISSUE TITLE
Django "OperationalError" when clearing recordstore (e.g. triggered by upgrading a Sumatra project from 0.6.0)

### ISSUE THREAD

**Issue by maxalbert** [open-research/sumatra#207]
Below are two ways to reproduce this. The first one demonstrates the simplest way of triggering the error. The the second one replicates upgrading a Sumatra project from a previous version, which is where I first encountered the error.

(i)

- Create a new Sumatra project:

   - `mkdir test_project && cd test_project`
   - `git init`
   - `smt init TestProject`

- In a Python session, run:

```python
>>> from sumatra.projects import load_project
>>> proj = load_project()
>>> proj.record_store.clear()
```

The last command throws the following exception:

```python
OperationalError                          Traceback (most recent call last)
<ipython-input-3-f2cf14e4f066> in <module>()
----> 1 proj.record_store.clear()

/home/vagrant/sumatra/sumatra/recordstore/django_store/__init__.pyc in clear(self)
    287         cur = connection.cursor()
    288         for cmd in cmds:
--> 289             cur.execute(cmd)
    290         db_config._create_databases()
    291 

/usr/local/lib/python2.7/dist-packages/django/db/backends/util.pyc in execute(self, sql, params)
     67         start = time()
     68         try:
---> 69             return super(CursorDebugWrapper, self).execute(sql, params)
     70         finally:
     71             stop = time()

/usr/local/lib/python2.7/dist-packages/django/db/backends/util.pyc in execute(self, sql, params)
     51                 return self.cursor.execute(sql)
     52             else:
---> 53                 return self.cursor.execute(sql, params)
     54 
     55     def executemany(self, sql, param_list):

/usr/local/lib/python2.7/dist-packages/django/db/utils.pyc in __exit__(self, exc_type, exc_value, traceback)
     97                 if dj_exc_type not in (DataError, IntegrityError):
     98                     self.wrapper.errors_occurred = True
---> 99                 six.reraise(dj_exc_type, dj_exc_value, traceback)
    100 
    101     def __call__(self, func):

/usr/local/lib/python2.7/dist-packages/django/db/backends/util.pyc in execute(self, sql, params)
     49         with self.db.wrap_database_errors:
     50             if params is None:
---> 51                 return self.cursor.execute(sql)
     52             else:
     53                 return self.cursor.execute(sql, params)

/usr/local/lib/python2.7/dist-packages/django/db/backends/sqlite3/base.pyc in execute(self, query, params)
    448     def execute(self, query, params=None):
    449         if params is None:
--> 450             return Database.Cursor.execute(self, query)
    451         query = self.convert_query(query)
    452         return Database.Cursor.execute(self, query, params)

OperationalError: no such table: django_store_record_output_data
```

----------------------------------------------------------

(ii) Here is an alternative way of triggering the error (when upgrading a Sumatra project):

- Install the last stable Sumatra version 0.6.0

- Create a new Sumatra project using this version (it does not even need any records). For example:

   - `mkdir test_project && cd test_project`
   - `git init`
   - `smt init TestProject`

- Export the project:

   `smt export`

- Install Sumatra version 0.7dev

- Try to upgrade the project to the latest Sumatra version:

   `smt upgrade`

The last command trigggers the exception:

```python
Traceback (most recent call last):
  File "/home/vagrant/sumatra/bin/smt", line 31, in <module>
    main(sys.argv[2:])
  File "/home/vagrant/sumatra/sumatra/commands.py", line 601, in upgrade
    project.record_store.clear()
  File "/home/vagrant/sumatra/sumatra/recordstore/django_store/__init__.py", line 289, in clear
    cur.execute(cmd)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 69, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 51, in execute
    return self.cursor.execute(sql)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/sqlite3/base.py", line 450, in execute
    return Database.Cursor.execute(self, query)
django.db.utils.OperationalError: no such table: django_store_record_output_data
```

**Comment by apdavison** [comment]
Method (i) triggers the error for me, but with method (ii) the upgrade is successful.

**Comment by maxalbert** [comment]
Thanks for trying to reproduce this! It's strange that you don't see the error with method (ii), since this is how I first stumbled across it when I tried to upgrade one of my projects. I have tested this in a virtual machine and can reproduce it there. Here is the ansible playbook which I used:

   https://gist.github.com/maxalbert/b19c40f2d01cf7cac631

There is probably no harm in trying to run this outside a VM (but perhaps comment out the first three tasks which are only needed to install the required packages). The command to run it is:

    ansible-playbook -c local -i localhost, reproduce_sumatra_issue_207.yml

It will clone Sumatra int ~/sumatra, check out version 0.6.0, create a test project in ~/test_project and will try to export the records and upgrade the project, which is when the failure happens.
